tiatoolbox/models/engine/io_config.py

```diff
@@ -0,0 +1,421 @@
+        ...     patch_input_shape=(224, 224),
+        ...     stride_shape=(224, 224),
+        ... )
+        >>> ioconfig = ioconfig.to_baseline()
+
+    """
+
+    input_resolutions: list[dict]
+    patch_input_shape: list[int] | np.ndarray | tuple[int, int]
+    stride_shape: list[int] | np.ndarray | tuple[int, int] = None
+    output_resolutions: list[dict] = field(default_factory=list)
+
+    def __post_init__(self):
+        """Perform post initialization tasks."""
+        if self.stride_shape is None:
+            self.stride_shape = self.patch_input_shape
+
+        self.resolution_unit = self.input_resolutions[0]["units"]
+
+        if self.resolution_unit == "mpp":
+            self.highest_input_resolution = min(
+                self.input_resolutions,
+                key=lambda x: x["resolution"],
+            )
+        else:
+            self.highest_input_resolution = max(
+                self.input_resolutions,
+                key=lambda x: x["resolution"],
+            )
+
+        self._validate()
+
+    def _validate(self):
+        """Validate the data format."""
+        resolutions = self.input_resolutions + self.output_resolutions
+        units = [v["units"] for v in resolutions]
+        units = np.unique(units)
+
+        if len(units) != 1:
+            msg = (
+                f"Multiple resolution units found: `{units}`. "
+                f"Mixing resolution units is not allowed."
+            )
+            raise ValueError(
+                msg,
+            )
+
+        if units[0] not in [
+            "power",
+            "baseline",
+            "mpp",
+        ]:
+            msg = f"Invalid resolution units `{units[0]}`."
+            raise ValueError(msg)
+
+    @staticmethod
+    def scale_to_highest(resolutions: list[dict], units: Units):
+        """Get the scaling factor from input resolutions.
+
+        This will convert resolutions to a scaling factor with respect to
+        the highest resolution found in the input resolutions list.
+
+        Args:
+            resolutions (list):
+                A list of resolutions where one is defined as
+                `{'resolution': value, 'unit': value}`
+            units (Units):
+                Resolution units.
+
+        Returns:
+            :class:`numpy.ndarray`:
+                A 1D array of scaling factors having the same length as
+                `resolutions`
+
+        """
+        old_val = [v["resolution"] for v in resolutions]
+        if units not in ["baseline", "mpp", "power"]:
+            msg = (
+                f"Unknown units `{units}`. "
+                f"Units should be one of 'baseline', 'mpp' or 'power'."
+            )
+            raise ValueError(
+                msg,
+            )
+        if units == "baseline":
+            return old_val
+        if units == "mpp":
+            return np.min(old_val) / np.array(old_val)
+        return np.array(old_val) / np.max(old_val)
```

Comment:
```suggestion
        old_vals = [v["resolution"] for v in resolutions]
        if units not in ["baseline", "mpp", "power"]:
            msg = (
                f"Unknown units `{units}`. "
                f"Units should be one of 'baseline', 'mpp' or 'power'."
            )
            raise ValueError(
                msg,
            )
        if units == "baseline":
            return old_vals
        if units == "mpp":
            return np.min(old_vals) / np.array(old_vals)
        return np.array(old_vals) / np.max(old_vals)
```

Reply:
Addressed in this commit https://github.com/TissueImageAnalytics/tiatoolbox/pull/618/commits/80ac0fcf5ae589b795a1667c79dc6096c7f2bf28